packaging/irods_prc_tests/test_rule_engine_plugin_metadata_guard_atomic.py

```diff
@@ -0,0 +1,126 @@
+        with iRODSSession(host='localhost', port=1247, user='rods', password='rods', zone='tempZone') as local_admin:
+            # create unprivileged user
+            local_admin.users.create_with_password('alice', 'test')
+
+            # unprivileged user creates test file
+            with iRODSSession(host='localhost', port=1247, user='alice', password='test', zone='tempZone') as unprivileged_user:
+                unprivileged_user.data_objects.create(cls.TEST_DATA_OBJECT_PATH)
+                unprivileged_user.acls.set( iRODSAccess('own', cls.TEST_DATA_OBJECT_PATH, 'rods', 'tempZone'))
+
+            coll = local_admin.collections.get("/tempZone")
+            # add metadata_guard config AVU to collection
+            METADATA_GUARD_CONFIG = {
+                'admin_only': True,
+                'prefixes': ['irods::']
+            }
+            if 'irods::metadata_guard' not in coll.metadata:
+                coll.metadata.add('irods::metadata_guard', json.dumps(METADATA_GUARD_CONFIG))
+
+            # create backup of irods config file
+            cls.backup_irods_config()
+            # insert metadata guard rule engine plugin to irods config file
+            with open(cls.IRODS_CONFIG_FILE_PATH, 'r+') as config_file:
+                METADATA_GUARD_PLUGIN_BLOCK = {
+                    'instance_name': 'irods_rule_engine_plugin-metadata_guard-instance',
+                    'plugin_name': 'irods_rule_engine_plugin-metadata_guard',
+                    'plugin_specific_configuration': {}
+                }
+                config_json = json.load(config_file)
+                config_json['plugin_configuration']['rule_engines'].insert(0, METADATA_GUARD_PLUGIN_BLOCK)
+            with open(cls.IRODS_CONFIG_FILE_PATH, 'wt') as config_file:
+                json.dump(config_json, config_file)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.restore_irods_config()
+
+    def test_bad_config(self):
+        # set metadata_guard config AVU on collection to nonsense
+        with iRODSSession(host='localhost', port=1247, user='rods', password='rods', zone='tempZone') as local_admin:
+            coll = local_admin.collections.get("/tempZone")
+            coll.metadata['irods::metadata_guard'] = iRODSMeta('irods::metadata_guard', '{"broken": ["irods::"], ')
+
+        with iRODSSession(host='localhost', port=1247, user='alice', password='test', zone='tempZone') as unprivileged_user:
+            obj = unprivileged_user.data_objects.get(self.TEST_DATA_OBJECT_PATH)
+            obj.metadata.apply_atomic_operations( AVUOperation(operation='add', avu=iRODSMeta('irods::atr1','val1')))
+
+            # bad config means operation should succeed
+            self.assertEqual(obj.metadata['irods::atr1'], iRODSMeta('irods::atr1', 'val1'))
+            obj.metadata.apply_atomic_operations( AVUOperation(operation='remove', avu=iRODSMeta('irods::atr1','val1')))
+
+    def test_guard_atomic_operations_admin_only(self):
```

Comment:
Append issue number to test name.

The convention for that is `<test_name>__issue_12345`.